scripts/pretraining/bert/create_pretraining_data.py

```diff
@@ -0,0 +1,532 @@
+# Copyright 2018 The Google AI Language Team Authors and DMLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Create masked LM/next sentence masked_lm examples for BERT."""
+
+
+import argparse
+import logging
+import io
+import os
+import glob
+import collections
+import warnings
+import random
+import time
+from multiprocessing import Pool
+import numpy as np
+import gluonnlp as nlp
+
+
+class TrainingInstance:
+    """A single training instance (sentence pair)."""
+
+    def __init__(self, tokens, segment_ids, masked_lm_positions,
+                 masked_lm_labels, is_random_next, vocab):
+        self.tokens = tokens
+        self.segment_ids = segment_ids
+        self.is_random_next = is_random_next
+        self.masked_lm_positions = masked_lm_positions
+        self.masked_lm_labels = masked_lm_labels
+        self.vocab = vocab
```

Comment:
May consider to remove the vocab from the TrainingInstance. You may also try to make it a dataclass (See SQuAD example: https://github.com/dmlc/gluon-nlp/blob/6c1ea59390772c0d9feac7d8fb03381d5db6d91c/scripts/question_answering/squad_utils.py#L59-L93)

Reply:
This is used for debugging (transforming the token id to the actual words in its string representation).